pkgs/development/libraries/gegl/default.nix

```diff
@@ -47,6 +47,10 @@ stdenv.mkDerivation rec {
     sha256 = "CkzbQWNeQGoISc0NPwPK99l8q4qhPShwfVMtAInVYSY=";
   };
 
+  patches = [
+    ./reproducibility_fix.patch
```

Comment:
Please add a comment what this does (e.g. `` Fixes `default-value` for `GeglConfig::threads` and `GeglConfig::tile-cache-size` properties in `Gegl-0.4.gir` ``) and a link to upstream merge request.

Reply:
There is no upstream merge request, because its pretty reasonable to tune the thread count/cache size to the machine running it, and it only becomes a problem when you care about reproducibility

Reply:
Well, reproducibility is not really a NixOS specific concern so it would better be addressed upstream.

And I would expect this to have serious effect on performance-critical code so in case, I think we should eschew reproducibility until we can do it in a way that does not affect performance.

Reply:
> its pretty reasonable to tune the thread count/cache size to the machine running it and it only becomes a problem when you care about reproducibility

Tuning to the thread count/cache size of the machine running it can definitely be helpful. When it's determined at compile-time, though (which is likely for most other distro's, not just NixOS), it's optimizing for the thread count/cache size of the build machine. That doesn't really seem helpful, regardless of whether you care about reproducibility :) .